### PR TITLE
En_Bombers

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Bombers/z_en_bombers.c [hunk 1]
```diff
@@ -33,45 +37,493 @@ const ActorInit En_Bombers_InitVars = {
...
+    &object_cs_Anim_0064B8, &object_cs_Anim_0053F4, &object_cs_Anim_01007C, &object_cs_Anim_00349C,
+    &object_cs_Anim_004960, &object_cs_Anim_005128, &object_cs_Anim_004C1C, &object_cs_Anim_001A1C,
+    &object_cs_Anim_003EE4, &object_cs_Anim_00478C, &object_cs_Anim_00433C, &object_cs_Anim_0060E8,
+    &object_cs_Anim_001708, &object_cs_Anim_005DC4, &object_cs_Anim_0026B0, &object_cs_Anim_0036B0,
+    &object_cs_Anim_0031C4,
+};
+
+u8 D_80C047E0[] = {
+    0, 0, 0, 2, 2, 0, 0, 0, 2, 0, 0, 0, 2, 0, 2, 0, 0, 0, 0, 0, 0,
+};
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bombers/z_en_bombers.c:70]:
```suggestion
u8 D_80C047E0[] = {
    ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP,
};
```
Based on usage

File: src/overlays/actors/ovl_En_Bombers/z_en_bombers.c [hunk 2]
```diff
@@ -33,45 +37,493 @@ const ActorInit En_Bombers_InitVars = {
...
+        case 0:
+            if (this->unk_2A8 == 0) {
+                Math_Vec3f_Copy(&sp60, &this->actor.home.pos);
+
+                sp60.x += randPlusMinusPoint5Scaled(150.0f);
+                sp60.z += randPlusMinusPoint5Scaled(150.0f);
+
+                abs = ABS_ALT(BINANG_SUB(this->actor.world.rot.y, Math_Vec3f_Yaw(&this->actor.world.pos, &sp60)));
+                if ((abs < 0x4000) && !BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.world.pos, &sp60, &sp6C,
+                                                               &colPoly, 1, 0, 0, 1, &sp48)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bombers/z_en_bombers.c:240]:
```suggestion
                if ((abs < 0x4000) && !BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.world.pos, &sp60, &sp6C,
                                                               &colPoly, true, false, false, true, &sp48)) {
```

File: src/overlays/actors/ovl_En_Bombers/z_en_bombers.c [hunk 3]
```diff
@@ -33,45 +37,493 @@ const ActorInit En_Bombers_InitVars = {
...
+                Math_SmoothStepToS(&this->actor.world.rot.y, this->unk_2A4, 0xA, 0x7D0, 0x14);
+            }
+
+            if ((s16)ABS_ALT(BINANG_SUB(this->actor.world.rot.y, this->unk_2A4)) < 0x100) {
+                Math_Vec3f_Copy(&sp54, &this->actor.world.pos);
+                sp54.x += Math_SinS(this->actor.world.rot.y) * 60.0f;
+                sp54.z += Math_CosS(this->actor.world.rot.y) * 60.0f;
+
+                if (BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.world.pos, &sp54, &sp6C, &colPoly, 1, 0, 0,
+                                            1, &sp48)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bombers/z_en_bombers.c:261]:
```suggestion
                if (BgCheck_EntityLineTest1(&globalCtx->colCtx, &this->actor.world.pos, &sp54, &sp6C, &colPoly, true, false, false,
                                            true, &sp48)) {
```

**Review by hensldm — APPROVED** [review]
Just some minor things.

File: src/overlays/actors/ovl_En_Bombers/z_en_bombers.c [hunk 4]
```diff
@@ -33,45 +37,493 @@ const ActorInit En_Bombers_InitVars = {
...
+                    ((this->unk_2BE == ENBOMBERS_F_3) && (gSaveContext.weekEventReg[74] & 4)) ||
+                    ((this->unk_2BE == ENBOMBERS_F_4) && (gSaveContext.weekEventReg[74] & 8))) {
+                    this->actor.textId = 0x74A;
+                }
+            } else {
+                this->actor.textId = 0x744;
+            }
+            break;
+    }
+}
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bombers/z_en_bombers.c:210]:
```suggestion
    switch (player->transformation) {
        case PLAYER_FORM_HUMAN:
            this->actor.textId = 0x73D;
            if (gSaveContext.weekEventReg[84] & 0x80) {
                this->actor.textId = 0x74B;
            }
            break;

        case PLAYER_FORM_GORON:
            this->actor.textId = 0x73E;
            break;

        case PLAYER_FORM_ZORA:
            this->actor.textId = 0x73F;
            break;

        case PLAYER_FORM_DEKU:
            if (gSaveContext.weekEventReg[73] & 0x20) {
                this->actor.textId = 0x75A;
            } else if (gSaveContext.weekEventReg[73] & 0x40) {
                this->actor.textId = 0x749;
                if (((this->unk_2BE == ENBOMBERS_F_1) && (gSaveContext.weekEventReg[74] & 1)) ||
                    ((this->unk_2BE == ENBOMBERS_F_2) && (gSaveContext.weekEventReg[74] & 2)) ||
                    ((this->unk_2BE == ENBOMBERS_F_3) && (gSaveContext.weekEventReg[74] & 4)) ||
                    ((this->unk_2BE == ENBOMBERS_F_4) && (gSaveContext.weekEventReg[74] & 8))) {
                    this->actor.textId = 0x74A;
                }
            } else {
                this->actor.textId = 0x744;
            }
            break;
    }
}
```

**Review by hensldm — COMMENTED** [review]
Noticed a couple things

**Review by AngheloAlf — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**